packages/task/src/node/task-server.ts

```diff
@@ -112,6 +126,28 @@ export class TaskServerImpl implements TaskServer, Disposable {
                             problems
                         });
                     }
+                    if (taskConfiguration.isBackground) {
+                        const backgroundTaskStatus = this.backgroundTaskStatusMap.get(event.taskId);
+
+                        const isBeginsPatternMatch = collector.matchBeginMatcher(event.line);
+                        if (isBeginsPatternMatch) {
+                            backgroundTaskStatus!.isBeginsPatternMatch = true;
+                        }
+
+                        // Get the 'activeOnStart' value of the problem matcher 'background' property
+                        const activeOnStart = collector.problemMatchers[0]!.watching!.activeOnStart;
+
+                        const isEndsPatternMatch = collector.matchEndMatcher(event.line);
+
+                        // Mark ends pattern as matches, only after begins pattern matches
+                        if (isEndsPatternMatch && (activeOnStart || backgroundTaskStatus!.isBeginsPatternMatch)) {
+                            backgroundTaskStatus!.isEndsPatternMatch = true;
+                            this.fireBackgroundTaskEndedEvent({
+                                taskId: event.taskId,
+                                ctx: event.ctx
+                            });
```

Comment:
I guess the latest change does not completely resolve one problem with `activeOnStart`.

```
		/**
		* If set to true the watcher is in active mode when the task
		* starts. This is equals of issuing a line that matches the
		* beginsPattern.
		*/
		activeOnStart?: boolean;
```

If you take a closer look at the source code of vscode, `activeOnStart = true` only makes the watch active when the task starts. In my opinion, the value of `activeOnStart` should not be taken into consideration in the if-statement (line 143) after the `BackgroundTaskEndedEvent` gets fired.

Here is an example that explains what I said:

- beginsPattern: code change detected, compiling
- endsPattern: changes compiled
- activeOnStart: true

```
compiling module 1 (start = false, end = false, activeOnStart = true)
compiling module 2 (start = false, end = false, activeOnStart = true)
changes compiled. no errors (start = false, end = true, activeOnStart = true, and we fire the event)
actively watching for more changes ... (start = false, end = false. From this point on, "activeOnStart = true" is obsolete, because the first "begin-end" pattern has been found)
```

Please correct me if I am wrong

Reply:
fixed, done

Reply:
quick question:
as you can see from the example in [this comment](https://github.com/eclipse-theia/theia/pull/6680#discussion_r355786903), the `beginsPattern - endsPattern` could be found multiple times in the task output. So ... how many `BackgroundTaskEndedEvent` are we going to fire? Do we just fire it on finding the first `beginsPattern - endsPattern`, or every time the `beginsPattern - endsPattern` is found ?

Reply:
When we first find the "**beginsPattern - endsPattern** " we fire the 'BackgroundTaskEndedEvent '.
And after the first 'BackgroundTaskEndedEvent ' was fired, for every time the  **endsPattern** is found we fire the 'BackgroundTaskEndedEvent '.